### PR TITLE
Migrated `jnosql-arangodb-extension` into the `jnosql-arangodb` module;

### PR DESCRIPTION
**Pull request by dearrudam**
Refs: https://github.com/eclipse/jnosql/issues/355

## Changes

- Migrated `jnosql-arangodb-extension` into the `jnosql-arangodb` module;

### CODE REVIEW

**Comment by dearrudam** [comment]
@otaviojava could you review this PR, please?

**Comment by genie-jnosql** [comment]
Can one of the admins verify this patch?

File: pom.xml [hunk 1]
```diff
@@ -103,6 +104,18 @@
                 <artifactId>jnosql-mapping-key-value</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
```
Comment (otaviojava) [inline pom.xml:107]: I've updated the pom parent; we don't need this anymore.
Could you check it, please?

**Comment by otaviojava** [comment]
@dearrudam please, update your branch to take the newest code from the main branch.

* Update the readme file with the documentation with Arangodb features: https://github.com/eclipse/jnosql-mapping-extension#how-to-install
*

**Comment by otaviojava** [comment]
About the package, please, use:
* `org.eclipse.jnosql.databases.arangodb.communication`
* `org.eclipse.jnosql.databases.arangodb.mapping`

:warning: this applies to putting the key-value and document types into a single package.

**Comment by dearrudam** [comment]
@otaviojava, please, review this one?!

File: jnosql-arangodb/src/main/java/org/eclipse/jnosql/databases/arangodb/mapping/AQL.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+package org.eclipse.jnosql.databases.arangodb.mapping;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * To a dynamic query on ArangoDBRepository and ArangoDBRepositoryAsync interfaces.
```
Comment (otaviojava) [inline jnosql-arangodb/src/main/java/org/eclipse/jnosql/databases/arangodb/mapping/AQL.java:23]: There is no `ArangoDBRepositoryAsync` could you update this one on another PR?

File: jnosql-arangodb/src/main/java/org/eclipse/jnosql/databases/arangodb/mapping/AQL.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface AQL {
+
+    String value();
```
Comment (otaviojava) [inline jnosql-arangodb/src/main/java/org/eclipse/jnosql/databases/arangodb/mapping/AQL.java:29]: on another PR, could you document this annotation?

**Review by otaviojava — APPROVED** [review]

**Pull request merged**